waspc/src/Wasp/Generator/DbGenerator.hs

```diff
@@ -43,13 +49,28 @@ dbSchemaFileInDbRootDir = SP.castRel dbSchemaFileInDbTemplatesDir
 dbSchemaFileInProjectRootDir :: Path' (Rel ProjectRootDir) File'
 dbSchemaFileInProjectRootDir = dbRootDirInProjectRootDir </> dbSchemaFileInDbRootDir
 
+dbMigrationsDirInDbRootDir :: Path' (Rel DbRootDir) (Dir DbMigrationsDir)
+dbMigrationsDirInDbRootDir = [reldir|migrations|]
+
 -- * Db generator
 
 genDb :: Wasp -> CompileOptions -> [FileDraft]
 genDb wasp _ =
   [ genPrismaSchema wasp
   ]
 
+preCleanup :: Wasp -> Path' Abs (Dir ProjectRootDir) -> CompileOptions -> IO ()
```

Comment:
This is looking good! Maybe you could create additional function called deleteGeneratedMigrationsDirIfRedundant and then call it in preCleanup as the only line in it, just to be a bit more verbose about clean up potentially doing more things and this being just one, clearly delineated action, but otherwise it all looks great.

There is one thing about naming that I would just like to emphasize but it has nothing to do with you right now -> it is the fact that we have this `waspMigrationsDir` vs `projectMigrationsDir`, where we use prefix `wasp` to indicate it is from the source directory with wasp code, while we use `project` to indicate it is from the generated project. Looking back at it I found it is a very confusing naming - "project" is too general, and "wasp" is also pretty lacking in information. It would probably be better if we used `srcMigrationsDir` and `genMigrationsDir`, and if we propagated this naming the whole codebase. Maybe this is ok place to start? Not sure hm. For exapmle `ProjectRootDir` -> that should probably be named `GenRootDir` or `GenProjectRootDir` or smth like that. Ok I digressed a bit :D. The only thing we might want to do here is use `src` and `gen` on new code, but I am not sure what is better - to start using new naming now and then have two different namings which we willl resolve bit by bit with time, or to stick to old naming until one day we do bigger renaming through the whole codebase. What do you think about the names btw?

Reply:
Good idea on the additional function to allow for more one-line cleanup items, will do!

And yes, totally agree on the naming. I originally had it called something like `src*` and `out*` but changed to match naming in this file. I was having a hard time following the naming convention myself across files/contexts, so I am glad you made that note. I think `src` and `gen` does convey more than `wasp` and `project`, so I am 💯  on board with those. I can update at least this file as the first step in that direction! 👍🏻

Reply:
Great! And in the future pls mention when you have hard time with a piece of code, so we can look into how to make it better!

Before we go full-on on renaming, we should probably make sure that the naming we choose makes sense.
For example, `gen` in front, although I suggested it just in previous comment :D, is (I see that now) also often used in front of function that generate sometihng, so that might be confusing (there are those functions just above - `genDb` `genPrismaSchema`). Hmmmmm. Maybe good starting point is to observe the names of the directories -> right now it is `ProjectRootDir` for generated project and `WaspProjectDir` for input/src project. Obviusly those are not very good names, since they are not even in sync. So maybe we should try to find new names for these, and that will then lead us to figuring out the other names also, is my hunch.
Hm now I am contradicting myself but I am actually thinking we should maybe not do any renaming in this PR because we will just end up introducing another naming scheme without thinking it through properly. Maybe we should leave it ias it is right now and instead create another issue where we will figure out naming, and once we do that then from that moment on we can use the new naming? WDYT?

Reply:
Sounds good on question front, thanks! At this stage, sometimes it is hard to know if my confusion is just being new to Haskell/a large project or "really worth discussing", but I will default to thinking more out loud on those things that give pause. :) 

Good thinking on creating a new issue to think this through more carefully vs rushing into a name for this PR we may decide wasn't the best in the long run. Overall, I do think having consistent terms/phrasing/conventions we like and can use in Docs + code would be amazing, and it will be good to think through more carefully before deciding! 👍🏻 

I will hold off on name changes here and create a new issue. Thanks!

Reply:
Awesome, let's do that! And yeah I assumed that was the reason you didn't mention this, but exactly as you said, feel free to be more vocal it can't hurt.

Reply:
Created the new issue here: https://github.com/wasp-lang/wasp/issues/396 Thanks!